lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def register(entity, map),
+  do: Entity.put_behaviour(entity, MapInstance.Behaviour, map)
+
+  def unregister(entity),
+  do: Entity.remove_behaviour(entity, MapInstance.Behaviour)
+
+  defmodule Behaviour do
+    use Entice.Entity.Behaviour
+
+    def init(entity, map),
+    do: {:ok, entity |> put_attribute(%MapInstance{map: map})}
+
+    def handle_call({:map_instance_player_join, player}, entity) when is_pid(player) do
+      add_creature_to_map_instance(entity, player, :players)
+    end
+
+    def handle_call({:map_instance_npc_join, npc}, entity) when is_pid(npc) do
```

Comment:
more like 'add an npc...' i think we should also actually create the npc entity here as well